package/MDAnalysis/analysis/hole2/hole.py

```diff
@@ -550,6 +550,8 @@ def __init__(self, universe,
             filenames.extend(universe.trajectory.filenames)
         except AttributeError:
             filenames.append(universe.trajectory.filename)
+        while None in filenames:
+            filenames.remove(None)
```

Comment:
Would a list generator be more efficient here, like `filenames = [x for x in filenames if x is not None]` ?

Reply:
yes, much more efficient :)